runtime/gc_base/accessBarrier.cpp

```diff
@@ -835,11 +837,47 @@ j9gc_objaccess_jniDeleteGlobalReference(J9VMThread *vmThread, J9Object *referenc
 	barrier->jniDeleteGlobalReference(vmThread, reference);
 }
 
-void
+j9object_t
+synchronizeWithConcurrentGCScan(J9VMThread *currentThread, j9object_t continuationObject)
+{
+	J9VMContinuation *continuation = J9VMJDKINTERNALVMCONTINUATION_VMREF(currentThread, continuationObject);
+	volatile uintptr_t *localAddr = &continuation->state;
+	/* atomically 'or' (not 'set') continuation->state  with currentThread */
+	uintptr_t oldContinuationState = VM_AtomicSupport::bitOr(localAddr, (uintptr_t)currentThread);
+
+	Assert_MM_true(NULL == VM_VMHelpers::getCarrierThreadFromContinuationState(oldContinuationState));
+
+	if (VM_VMHelpers::isConcurrentlyScannedFromContinuationState(oldContinuationState)) {
+		/* currentThread was low tagged (GC was already in progress), but by 'or'-ing our ID, we let GC know there is a pending mount */
+
+		PUSH_OBJECT_IN_SPECIAL_FRAME(currentThread, continuationObject);
+		currentThread->javaVM->internalVMFunctions->internalReleaseVMAccess(currentThread);
+
+		omrthread_monitor_enter(currentThread->publicFlagsMutex);
+		while (VM_VMHelpers::isConcurrentlyScannedFromContinuationState(*localAddr)) {
+			/* GC is still concurrently scanning the continuation(currentThread was still low tagged), wait for GC thread to notify us when it's done. */
+			omrthread_monitor_wait(currentThread->publicFlagsMutex);
+		}
+		omrthread_monitor_exit(currentThread->publicFlagsMutex);
+
+		currentThread->javaVM->internalVMFunctions->internalAcquireVMAccess(currentThread);
+		continuationObject = POP_OBJECT_IN_SPECIAL_FRAME(currentThread);
+	}
+
+	Assert_MM_true((uintptr_t)currentThread == continuation->state);
+	return continuationObject;
+}
+
+
+j9object_t
 preMountContinuation(J9VMThread *vmThread, j9object_t object)
 {
+	/* let GC know we are mounting, so they don't need to scan us, or if there is already ongoing scan wait till it's complete. */
+	object = synchronizeWithConcurrentGCScan(vmThread, object);
```

Comment:
once we move it, I would prefer to move it within the Barrier base class
this is just a simple C wrapper, as with all others in this file - we don't expect real function code here, other then being a wrapper